portfolio/src/main/webapp/script.js

```diff
@@ -31,16 +31,82 @@ function addRandomGreeting() {
  * Opens a modal with the given image and caption.
  */
 function openSimpleImageModal(modalId, modalImgId, captionId, closeId, imgSrc, captionText) {  
-  var modal = document.getElementById(modalId);
-  var modalImg = document.getElementById(modalImgId);
-  var caption = document.getElementById(captionId);
+  const modal = document.getElementById(modalId);
+  const modalImg = document.getElementById(modalImgId);
+  const caption = document.getElementById(captionId);
 
   modal.style.display = "block";
   modalImg.src = imgSrc;
-  caption.innerHTML = captionText;
+  caption.innerText = captionText;
 
-  var closeBtn = document.getElementById(closeId);
+  const closeBtn = document.getElementById(closeId);
   closeBtn.onclick = function() {
     modal.style.display = "none";
   };
 }
+
+/**
+ * Advance or Reverse the slide currently showing.
+ */
+function advanceSlides(n) {
+  currIndex += n;
+  currIndex = showSlidePicture(currIndex);
+};
+
+/**
+ * Show the image from the gallery at the given position.
+ */
+function showSlidePicture(n) {
+  const slides = document.getElementsByClassName("slide-pic");
+  const caption = document.getElementById("slide-caption");
+
+  // If not on pics.html document wil not contain any slide elements.
+  if (slides.length === 0) {
+    return n;
+  }
+
+  let newIndex = circularIndex(n, slides);
+  
+  for (let slide of slides) {
+    slide.style.display = "none";
+  }
+
+  slides[newIndex].style.display = "block";
+  slides[newIndex].getElementsByClassName("slide-counter")[0].innerText =
+     (newIndex + 1).toString() + "/" + slides.length.toString();
+  caption.innerText = picCaptions[newIndex];
+
+  return newIndex;
+};
+
+/**
+ * Return correct index for a circular array.
+ */
+function circularIndex(n, items) {
+  while (n < 0) {
+    n += items.length;
+  }
+
+  return n % items.length;
+}
+
+/**
+ * Initialise map for currently displayed image.
+ */
+function initMap() {
+
+}
+
+let picCaptions = [
```

Comment:
Images are disconnected from captions. I think it would be better to either:
* have a single array with elements with 2 fields: src for the img and the caption
* put caption directly to html, inside slide-pic

Otherwise it's easy to accidentally reorder them, or add a new picture without adding its caption

Reply:
Good point, I have grouped the data to now be in a single array of objects, which can be future expanded depending on any other data that needs to be stored, or other images need to be added. To accommodate for this, HTML for the images is now generated using createElement on load of the page.